languages/csharp/exercises/concept/parameters/.docs/instructions.md

```diff
@@ -0,0 +1,80 @@
+The remote control car project you kicked off in the `classes` exercise has gone well (congratulations!) and due to a number of recent sponsorship deals there is money in the budget for enhancements.
+
+Part of the budget is being used to provide some telemetry.
+
+To keep the sponsors sweet a panel has been installed on the car to display the sponsors names as it goes round the track.
+
+You will note that the introduction of these fancy new technologies has dramatically reduced the car's battery life.
+
+## 1. Enable the remote control car to display sponsor names
+
+Implement `AddSponsors` to take one or more sponsor names and store them on the car.
+
+Note `AddSponsors` is guaranteed to be called at most once and the argument is guaranteed to be non-null.
+
+Implement `DisplaySponsor` to display the selected sponsor. The first sponsor added has a `sponsorNum` of 0, the second, 1 etc.
+
+```csharp
+var car = RemoteControlCar.Buy();
+car.AddSponsors("Exercism", "Walker Industries", "Acme Co.");
+var sp2 = car.DisplaySponsor(sponsorNum: 1);
+// => "Walker Industries"
+```
+
+## 2 Get the car's telemetry data
+
+Implement `RemoteControlCar.GetTelemetryData()`.
+
+`GetTelemetryData()` should make the battery percentage and distance driven in meters available via `out` parameters.
+
+`GetTelementryData()` should return `false` if the timestamp argument is less than the previously received value. (There is some issue of multiple telemetry nodes being involved). In this case the timestamp will be set to the highest previous timestamp and battery percentage and meters driven will both be set to `-1`. Where the call is successful the timestamp remains unchanged.
+
+```csharp
+var car = RemoteControlCar.Buy();
+car.Drive();
+car.Drive();
+long timestamp = 4L;
```

Comment:
Use a `long` value does mean taking a dependency on the `integral-numbers` concept. It does fit the timestamp model of course. An alternative would be a simple `int` follow-up number that represents the nth telemetry data recording. I don't mind keeping the long if you prefer that.

Reply:
I have substituted `int serialNum`.

Reply:
👍